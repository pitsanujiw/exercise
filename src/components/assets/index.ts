export * from './assetItem';
export * from './assetsTable';
