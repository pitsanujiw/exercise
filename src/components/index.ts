export * from "./assets";
export * from "./loading";
export * from "./orderForm";
