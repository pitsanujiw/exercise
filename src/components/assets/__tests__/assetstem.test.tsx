import { AssetItem } from "components";
import { Table, TableBody } from "@material-ui/core";
import { render } from "@testing-library/react";

describe("Asset Item", () => {
  it("Should render it correctly", async () => {
    const renderResult = render(
      <div>
        <Table>
          <TableBody>
            <AssetItem
              asset={{
                price: "10",
                amount: "10",
              }}
              type="sell"
            />
          </TableBody>
        </Table>
      </div>
    );
    expect(renderResult).toMatchSnapshot();
  });
});
