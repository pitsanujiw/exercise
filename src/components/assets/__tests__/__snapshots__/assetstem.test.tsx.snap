// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Item Should render it correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root AssetItem-table-5 AssetItem-asset-1 AssetItem-down-3"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
              >
                $10.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
              >
                $10.00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="MuiTable-root"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root AssetItem-table-5 AssetItem-asset-1 AssetItem-down-3"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
            >
              $10.00
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
            >
              $10.00
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
